photoeditor/src/main/java/ja/burhanrashid52/photoeditor/PhotoEditor.java

```diff
@@ -47,13 +49,45 @@
     private ImageView imageView;
     private View deleteView;
     private BrushDrawingView brushDrawingView;
-    private List<View> addedViews;
-    private List<View> redoViews;
     private OnPhotoEditorListener mOnPhotoEditorListener;
     private boolean isTextPinchZoomable;
     private Typeface mDefaultTextTypeface;
     private Typeface mDefaultEmojiTypeface;
+    private ViewState viewState;
 
+    // Tracked state of user-added views (stickers, emoji, text, etc)
+    public static class ViewState {
```

Comment:
We need to move this `ViewState` class out of the `PhotoEditor` because it getting bigger.

Reply:
Done